### PR TITLE
New Room List accessibility

### PR DESCRIPTION
**Pull request by t3chguy**
For https://github.com/vector-im/riot-web/issues/13635
For https://github.com/vector-im/riot-web/issues/14180

### CODE REVIEW

File: src/components/structures/ContextMenu.js [hunk 1]
```diff
@@ -421,6 +423,54 @@ MenuItemCheckbox.propTypes = {
     onClick: PropTypes.func.isRequired,
 };
 
+// Semantic component for representing a styled role=menuitemcheckbox
+export const StyledMenuItemCheckbox = ({children, label, onChange, onClose, checked, disabled=false, ...props}) => {
```
Comment (turt2live) [inline src/components/structures/ContextMenu.js:427]: as a distant future problem: we should consider not allowing multiple components in the same file, as this file is illegible.

File: src/components/views/rooms/RoomTile2.tsx [hunk 2]
```diff
@@ -68,6 +80,10 @@ interface IState {
     generalMenuPosition: PartialDOMRect;
 }
 
+type Volume = ALL_MESSAGES_LOUD | ALL_MESSAGES | MENTIONS_ONLY | MUTE;
```
Comment (turt2live) [inline src/components/views/rooms/RoomTile2.tsx:83]: This should be defined elsewhere, not here.

Comment (t3chguy) [reply src/components/views/rooms/RoomTile2.tsx:83]: There is currently no sane place to put it, RoomNotifs is JS

Comment (turt2live) [reply src/components/views/rooms/RoomTile2.tsx:83]: we can do something similar to the early dispatcher work: `RoomNotifTypes.ts` with a comment saying `// TODO: The rest`.

Comment (t3chguy) [reply src/components/views/rooms/RoomTile2.tsx:83]: RoomNotif is probably dying as part of FTUE Notifications though :((

Comment (turt2live) [reply src/components/views/rooms/RoomTile2.tsx:83]: `Volume.ts` then? It's just too wrong to leave it here (it's bad enough there's membership handling utils in the room list store, we shouldn't really be pushing bad patterns through)

Comment (t3chguy) [reply src/components/views/rooms/RoomTile2.tsx:83]: Volume as a concept is being nuked then but sure will place it somewhere.

File: src/RoomNotifsTypes.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+*/
+
+import {
+    ALL_MESSAGES,
+    ALL_MESSAGES_LOUD,
+    MENTIONS_ONLY,
+    MUTE,
+} from "./RoomNotifs";
+
+export type Volume = ALL_MESSAGES_LOUD | ALL_MESSAGES | MENTIONS_ONLY | MUTE;
```
Comment (turt2live) [inline src/RoomNotifsTypes.ts:24]: feel free to litter this with comemnts to either A: remove it or B: update it

Comment (t3chguy) [reply src/RoomNotifsTypes.ts:24]: no need, as part of FTUE Notificaitons the underlying ALL_MESSAGES_LOUD etc will be going away and thus it'll cause build errors anyhow :D

**Pull request merged**